activation_key_logger_test.go

```diff
@@ -0,0 +1,457 @@
+package micrologger
+
+import (
+	"testing"
+)
+
+func Test_ActivationKeyLogger_shouldActivate_zeroValue(t *testing.T) {
+	testCases := []struct {
+		Activations    map[string]interface{}
+		KeyVals        []interface{}
+		ExpectedResult bool
+	}{
+		// Case 0, zero value input results into false, because logging should not
+		// be activated in case no match exists, even if the input is empty.
+		{
+			Activations:    nil,
+			KeyVals:        nil,
+			ExpectedResult: false,
+		},
+
+		// Case 1, same as 0 but with empty lists instead of zero values.
+		{
+			Activations:    map[string]interface{}{},
+			KeyVals:        []interface{}{},
+			ExpectedResult: false,
+		},
+
+		// Case 2, same as 0 but with different input.
+		{
+			Activations:    nil,
+			KeyVals:        []interface{}{},
+			ExpectedResult: false,
+		},
+
+		// Case 3, same as 0 but with different input.
+		{
+			Activations:    map[string]interface{}{},
+			KeyVals:        nil,
+			ExpectedResult: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		result, err := shouldActivate(tc.Activations, tc.KeyVals)
+		if err != nil {
+			t.Fatalf("case %d expected %#v got %#v", i, nil, err)
```

Comment:
Wouldn't calling `t.Fatalf` without running it inside `t.Run() {}` interrupt the whole test function here, skipping rest of the remaining test cases? Is this intentional here?

Reply:
I never used something else than the fatal magic. When a test fails, it stops execution. Other tests are not affected by this. I am also not used to the usage of the sub tests but I am glad to experiment and get in touch with this in the future. I just feel like we are still in the middle of finding out how to use this.